.eslintrc

```diff
@@ -115,5 +116,37 @@
     "When": true,
     "Then": true,
     "And": true
-  }
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "ecmaFeatures": {
+          "jsx": true
+        },
+        "project": "./tsconfig.json"
+      },
+      "plugins": ["@typescript-eslint", "jest"],
+      "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:@typescript-eslint/recommended",
+        "prettier",
+        "prettier/@typescript-eslint",
+        "plugin:react-hooks/recommended",
+        "plugin:jest/recommended",
+        "plugin:jest-dom/recommended"
+      ],
+      "settings": {
+        "react": {
+          "version": "detect"
+        }
+      },
+      "rules": {
```

Comment:
Why are we turning these rules off? We don't have any TS components yet so these shouldn't be failing?

Reply:
The `import/named` one because it errors for every import we have in `.d.ts` files - I believe we have actually implemented it incorrectly but converting Carbon to TypeScript will fix it

The second one because it errors for all our class components, if they don't have state - we can turn this back on when we've done all the functional refactor tickets.

I suppose for both we could disable the rules line by line where needed but I thought it wasn't worth it, and fully intend on turning these rules back on once the TypeScript conversion and class -> functional tickets are all done